accepted/future-releases/base-interface-final/feature-specification.md

```diff
@@ -0,0 +1,596 @@
+the class to be used as a mixin by adding a `mixin` modifier to the class:
+
+```dart
+class OnlyClass {}
+
+class FailUseAsMixin extends OtherSuperclass with OnlyClass {} // Error.
+
+mixin class Both {}
+
+class UsesAsSuperclass extends Both {}
+
+class UsesAsMixin extends OtherSuperclass with Both {} // OK.
+```
+
+This change is guarded by a language version. Any class declaration in a library
+whose language version is before the one this feature ships in can be used as a
+mixin as long as the class meets the mixin restrictions. Classes in libraries
+whose version is later than that must be explicitly marked `mixin class` to
+allow the class to be used in a `with` clause.
+
+When upgrading your library to the new language version, you can preserve the
+existing behavior by adding `mixin` to every class declaration that can be used
+as a mixin. If the class defines a generative constructor or extends anything
+other than `Object`, then it already can't be used as a mixin and no change is
+needed.
+
+## Syntax
+
+This proposal builds on the existing sealed types proposal so the grammar
+includes those changes. The full set of modifiers that can appear before a class
+or mixin are `abstract`, `sealed`, `base`, `interface`, `final`, and `mixin`.
+Some combinations don't make sense:
+
+*   `sealed` implies `abstract`, so they can't be combined.
+*   `sealed` implies non-extensibility, so can't be combined with `interface`
+    or `final`.
+*   `sealed` implies non-implementability, so can't be combined with `base` or
+    `final`.
+*   `base`, `interface`, and `final` all control the same two capabilities so
+    are mutually exclusive.
+*   `mixin` as a modifier can obviously only be applied to a class. It can be
+    combined with any other modifiers that can be applied to a class.
+
+The remaining valid combinations are:
+
+<table><tr><td>
+
+```
+class
+sealed class
+base class
+interface class
+final class
+abstract class
+abstract base class
+abstract interface class
+abstract final class
+```
+
+</td><td>
+
+```
+mixin class
+sealed mixin class
+base mixin class
+interface mixin class
+final mixin class
+abstract mixin class
+abstract base mixin class
+abstract interface mixin class
+abstract final mixin class
+```
+
+</td><td>
+
+```
+mixin
+sealed mixin
+base mixin
+interface mixin
+final mixin
+```
+
+</td></tr></table>
+
+The grammar is:
+
+```
+classDeclaration ::=
+  classModifiers 'class' identifier typeParameters?
+  superclass? interfaces?
+  '{' (metadata classMemberDeclaration)* '}'
+  | classModifiers 'class' mixinApplicationClass
+
+mixinDeclaration ::= mixinModifier? 'mixin' identifier typeParameters?
+  ('on' typeNotVoidList)? interfaces?
+  '{' (metadata classMemberDeclaration)* '}'
+
+classModifiers ::= ( 'sealed' | 'abstract'? typeModifier? ) 'mixin'?
+mixinModifier ::= 'sealed' | typeModifier
+typeModifier ::= 'base' | 'interface' | 'final'
+```
+
+### Static semantics
+
+It is a compile-time error to:
```

Comment:
The syntax allows the modifiers to be applied to a `mixin` declaration, but there are no compile time errors specified for mixins below.  Presumably there is something intended here, but I don't actually know what.

Reply:
Fixed.